File: src/main/java/co/com/sofka/gestionriesgos/collections/Risk.java
package co.com.sofka.gestionriesgos.collections;

import lombok.*;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "risks")
public class Risk {
    @Id
    private String id;

    private String projectId;

    private String name;

    private String userId;

    private LocalDate detectedDate = LocalDate.now();

    private LocalDate endedDate;

    private List<String> labels;

    private String description;

    // (Abierto; mitigado; Cerrado; Problema)
    private String riskState;

    // (Interna; Externa)
    private String audience;

    // (Costo; Tiempo; Calidad)
    private String category;

    // (Riesgo de producto o calidad; Riesgo de proyecto)
    private String riskType;

    private String detailsRiskType;

    private Integer probability;

    private Integer impactValue;

    private String mitigationPlan;

    private List<String> responsibleMitigationMails;

    private String contingencyPlan;

    private List<String> responsibleContingencyMails;

    private Integer state = 1;
}

File: web/src/app/models/risk.ts
export interface Risk {
  id: string;
  projectId: string;
  name: string;
  userId: string;
  detectedDate: Date;
  endedDate: Date;
  labels: [string];
  description: string;
  // (Abierto; mitigado; Cerrado; Problema)
  risksState: string;
  // (Interna; Externa)
  audience: string;
  // (Costo; Tiempo; Calidad)
  category: string;
  // (Riesgo de producto o calidad; Riesgo de proyecto)
  riskType: string;
  detailsRiskType: string;
  probability: number;
  impactValue: number;
  mitigationPlan: string;
  responsibleMitigationMails: [string];
  contingencyPlan: string;
  responsibleContingencyMails: [string];
  state: number;
}
